src/ca/ubc/cs304/controller/MakeReservationController.java

```diff
@@ -21,7 +22,30 @@ public MakeReservationController(JFrame currentWindow) {
 
     @Override
     public void createReservation(Reservation reservation, Branch branch) {
-        if (!checkDateIsValid(reservation.getFromDate(), reservation.getToDate())) {
+        System.out.println("fromDate:"+reservation.getToTime());
+
+        if (!checkDateIsValid(reservation.getFromDate(), reservation.getToDate(), reservation.getFromTime(), reservation.getToTime())) {
+            currentWindow.dispose();
+            ErrorWindow errorWindow = new ErrorWindow();
+            errorWindow.infoBox("Sorry, time information is invalid", "Invalid Input!");
+            MakeReservationWindow makeReservationWindow = new MakeReservationWindow();
+            MakeReservationController makeReservationController = new MakeReservationController(makeReservationWindow);
+            ArrayList<String> vtlist = dbHandler.findAllVT();
```

Comment:
you can move these to the top , above the if statement, and won't have to repeat so much code